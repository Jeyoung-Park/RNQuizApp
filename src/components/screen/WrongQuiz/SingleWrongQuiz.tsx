import React, { useState } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import SingleQuiz from '../Quiz/SingleQuiz';

interface SingleWrongQuizProps {
	navigation: any;
	route: any;
}

const SingleWrongQuiz = ({ navigation, route }: SingleWrongQuizProps) => {
	const { quizItem } = route.params;

	const [isSolved, setIsSolved] = useState<boolean>(false);

	return (
		<View style={styles.container}>
			{/* <Text>SingleWrongQuiz</Text> */}
			<View style={styles.singleQuizContainer}>
				<SingleQuiz
					currentQuiz={quizItem}
					isSolved={isSolved}
					setIsSolved={(param) => setIsSolved(param)}
				/>
			</View>
		</View>
	);
};

export default SingleWrongQuiz;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: 'white',
	},
	singleQuizContainer: {
		flex: 1,
	},
});
