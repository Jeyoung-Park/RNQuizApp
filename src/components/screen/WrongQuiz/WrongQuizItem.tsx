import * as React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import { decode } from 'html-entities';
import { Quiz } from '../../../types';

interface WrongQuizItemProps {
	quizItem: Quiz;
	itemHeight: number;
	onPress: (param: Quiz) => void;
}

const WrongQuizItem = ({
	quizItem,
	itemHeight,
	onPress,
}: WrongQuizItemProps) => (
	<TouchableOpacity
		style={[styles.container, { height: itemHeight }]}
		onPress={() => onPress(quizItem)}
	>
		<Text>{decode(quizItem.question)}</Text>
	</TouchableOpacity>
);

export default WrongQuizItem;

const styles = StyleSheet.create({
	container: {
		width: '100%',
		// height: 80,
		justifyContent: 'center',

		paddingHorizontal: 20,

		borderBottomColor: 'gray',
		borderBottomWidth: 1,
	},
});
