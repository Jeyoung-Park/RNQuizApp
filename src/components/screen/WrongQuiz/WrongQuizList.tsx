import React from 'react';
import { FlatList, StyleSheet, Text, View } from 'react-native';
import { useSelector } from 'react-redux';
import wrongQuiz from '../../../models/wrongQuiz';

const WrongQuizList = () => {
	const wrongQuizList = useSelector((state) => state.wrongQuiz.wrongQuizList);

	console.log('wrongQuizList in WrongQuizList, ', wrongQuizList);

	return (
		<FlatList
			data={wrongQuizList}
			keyExtractor={(item, index) => item.question}
			renderItem={({ item, index }) => <Text>{item.question}</Text>}
			ListEmptyComponent={() => <Text>오답이 없습니다.</Text>}
			contentContainerStyle={
				wrongQuizList.length === 0 && styles.emptyComponent
			}
		/>
	);
};

export default WrongQuizList;

const styles = StyleSheet.create({
	emptyComponent: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
	},
});
