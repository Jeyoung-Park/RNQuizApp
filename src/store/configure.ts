import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../models';

const configure = () => createStore(reducer, applyMiddleware(thunk));

export default configure;
