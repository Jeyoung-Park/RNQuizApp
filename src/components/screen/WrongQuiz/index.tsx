/* eslint-disable @typescript-eslint/no-use-before-define */
import * as React from 'react';
import { Text, View, StyleSheet, SafeAreaView } from 'react-native';
import Header from './Header';
// import WrongQuizList from './WrongQuizList';
import WrongQuizList from './WrongQuizList';

interface WrongQuizProps {
	navigation: any;
}

const WrongQuiz = ({ navigation }: WrongQuizProps) => {
	const goBack = () => {
		navigation.goBack();
	};

	return (
		<SafeAreaView style={styles.container}>
			<Header goBack={goBack} />
			<WrongQuizList />
		</SafeAreaView>
	);
};

export default WrongQuiz;

const styles = StyleSheet.create({
	container: {
		flex: 1,
	},
});
