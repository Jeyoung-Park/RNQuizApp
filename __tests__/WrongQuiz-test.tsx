import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import store from '../src/store';
import WrongQuiz from '../src/components/screen/WrongQuiz';

jest.mock('react-native-simple-toast', () => ({
	SHORT: jest.fn(),
}));

it('snapshot test', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<WrongQuiz />
			</Provider>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
