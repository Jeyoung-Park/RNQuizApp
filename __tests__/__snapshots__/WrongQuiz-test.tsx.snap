// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      오답 노트
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    />
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={Object {}}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <Text>
        오답이 없습니다.
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
